gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/TimestampBasedCopyableDataset.java

```diff
@@ -73,6 +71,7 @@
   private final VersionSelectionPolicy<TimestampedDatasetVersion> versionSelectionPolicy;
   private final ExecutorService executor;
   private final FileSystem srcFs;
+  private final CopyableFileFilter copyableFileFilter;
```

Comment:
This class already has a method copyableFileFilter() that returns a HiddenFilter. You can use AndPathFilter to merge this filter with the filter specified in member variable copyableFileFilter.

Reply:
PathFilter interfaces do not support operation to merge the filter, also before filtering through the data range filter, hidden files are removed from the list by the existing control flow.

Reply:
Please take a look at AndPathFilter in gobblin-utility and UnixTimestampRecursiveCopyableDataset for an example usage.

Reply:
Thanks for the reference. AndPathFilter and CopyableFileFilter are two different interfaces and did not find a way to merge. AndPathFilter implements accept(..) whereas CopyableFileFilter implements filter(..). Please advise.